apps/backend/src/mcp/tools/device_info.py

```
@@ -0,0 +1,1204 @@
+from apps.backend.src.models.device import Device
+from apps.backend.src.utils.ssh_client import get_ssh_client, SSHConnectionInfo, execute_ssh_command_simple
+from apps.backend.src.core.exceptions import (
+    DeviceNotFoundError, SSHConnectionError, SystemMonitoringError
+)
+from apps.backend.src.core.config import get_settings
+
+logger = logging.getLogger(__name__)
+
+
+async def _collect_smart_data(
+    ssh_client, 
```

Ruff (0.12.2)

Trailing whitespace

Remove trailing whitespace

(W291)